lib/thread/no_os/semaphore.cpp

```diff
@@ -69,16 +86,26 @@ void ecl::binary_semaphore::wait()
 
 bool ecl::binary_semaphore::try_wait(std::chrono::milliseconds ms)
 {
+
+    bool ex;
+
+#ifdef USE_SYSTMR
```

Comment:
Actually, `wait_for` is pretty rubbish when used with `clk()` interface. Moreover, since `clk*` interfaces are so imprecise and current clock counter is overflow-prone, the clock usage will be dropped from any time-related routines. The clock interfaces thus will fall into optional category for arch/platform. I'll add ticket for that and put some comments.